src/model_training.py

```diff
@@ -1,53 +1,117 @@
+from dataclasses import asdict
+from datetime import datetime
+
 import tensorflow as tf
 import os
+import logging
+
+from src.constants import LOG_FILE_NAME
 from src.data import BaldDataset
+from src.logging import check_if_log_exists
 from src.model import BaldOrNotModel
-from src.config import BoldOrNotConfig
+from src.config_class import BoldOrNotConfig
 
 
-def train_model(config: BoldOrNotConfig):
+def train_model(config: BoldOrNotConfig, output_dir_path: str):
     """
     Trains the BaldOrNot model using the specified configuration.
 
-    This function initializes the dataset, constructs the model, compiles it with the specified
-    optimizer, loss function, and metrics, and then trains the model on the dataset for the
-    number of epochs defined in the configuration.
+    This function initializes the dataset, constructs the model, compiles it
+    with the specified optimizer, loss function, and metrics, and then trains
+    the model on the dataset for the number of epochs defined in the
+    configuration.
 
     Args:
-        config (BoldOrNotConfig): The configuration object containing model, training, and path parameters.
+        config (BoldOrNotConfig): The configuration object containing model,
+        training, and path parameters.
     """
+    if not check_if_log_exists(output_dir_path):
+        raise RuntimeError(
+            "Log file '{}' not found in '{}'. Make sure logging "
+            "is initialized.".format(LOG_FILE_NAME, output_dir_path)
+        )
```

Comment:
Create a decorator: https://blog.hubspot.com/website/decorators-in-python
put it in utils